test/integration/cdi/example-driver/kubeletplugin/kubeletplugin.go

```diff
@@ -0,0 +1,135 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	cdiapi "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	specs "github.com/container-orchestrated-devices/container-device-interface/specs-go"
+	"k8s.io/klog/v2"
+	drapbv1 "k8s.io/kubernetes/pkg/kubelet/apis/dra/v1alpha1"
+	server "k8s.io/kubernetes/test/integration/cdi/example-driver/kubeletplugin/server"
+)
+
+type PluginConfig struct {
+	DriverName             string
+	DraAddress             string
+	PluginRegistrationPath string
+}
+
+type examplePlugin struct {
+	config PluginConfig
+}
+
+// newExamplePlugin returns an initialized examplePlugin instance
+func newExampleDriver(config PluginConfig) (*examplePlugin, error) {
+
+	if err := os.MkdirAll(filepath.Dir(config.DraAddress), 0750); err != nil {
+		return nil, err
+	}
+
+	return (&examplePlugin{
+		config: config,
+	}), nil
+}
+
+func StartPlugin(config *PluginConfig) error {
+
+	err := server.StartRegistrar(config.DriverName, config.DraAddress, config.PluginRegistrationPath)
+	if err != nil {
+		return err
+	}
+
+	driver, err := newExampleDriver(*config)
+	if err != nil {
+		return err
+	}
+
+	s := server.NewNonBlockingGRPCServer()
+	// ex itself implements NodeServer.
+	s.Start(config.DraAddress, driver)
+	s.Wait()
+
+	return nil
+}
+
+var jsonFilePath = "/etc/cdi/vendor.json"
+var devicePath = "/dev/tty3"
+var cdiVersion = "0.2.0"
+var kind = "vendor.com/device"
+var deviceName = "example"
+
+func (ex *examplePlugin) NodePrepareResource(ctx context.Context, req *drapbv1.NodePrepareResourceRequest) (*drapbv1.NodePrepareResourceResponse, error) {
+	klog.Infof("NodePrepareResource is called: request: %+v", req)
+	// create CDI Files
+
+	if _, err := os.Stat(devicePath); os.IsNotExist(err) {
+		klog.Errorf(err.Error())
+		return &drapbv1.NodePrepareResourceResponse{}, err
+	} else {
+		// create spec
+		klog.Infof("Creating CDI Files")
+
+		containerEditsInDevices := &cdiapi.ContainerEdits{}
+		containerEditsInDevices.Append(&cdiapi.ContainerEdits{ContainerEdits: &specs.ContainerEdits{
```

Comment:
Instead of hard-coding some content, decode the handle string into a string/string map and set that as environment variables. That's what has been documented as functionality of the example driver in the README and what the controller implements.

Reply:
Maybe done.! If you think it's done. Please, resolve it.! I am not sure.